alvr/dashboard/src/data_sources.rs

```diff
@@ -128,6 +133,72 @@ impl DataSources {
         let port = session_manager.settings().connection.web_server_port;
         let session_source = Arc::new(Mutex::new(SessionSource::Local(Box::new(session_manager))));
 
+        let version_check_thread = thread::spawn({
+            let context = context.clone();
+            let session_source = Arc::clone(&session_source);
+            let events_sender = events_sender.clone();
+            move || {
+                let version_requirement = {
+                    // Best-effort: the check will succeed only when the server is not already running,
+                    // no retries
+                    let SessionSource::Local(session_manager_lock) = &mut *session_source.lock()
+                    else {
+                        return;
+                    };
```

Comment:
Can't u just return an error to not have to do that every time? I don't think spawn actually cares abt whether there's an error

Reply:
I don't quite understand what you are talking about. You mean using the `?` operator? That is only supported by `Option` and `Result` but here I'm branching with the custom `SessionSource` enum.

Reply:
I selected the wrong one, but maybe consider that for the instances here where there's actually Options and Results being handled (if u can make it work by just returning a result from the thread that u just ignore)

Reply:
Alright, sure